core/transaction_pool/transaction_pool.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_TRANSACTION_POOL_HPP
+#define KAGOME_TRANSACTION_POOL_HPP
+
+#include <outcome/outcome.hpp>
+#include "primitives/block_id.hpp"
+#include "primitives/transaction.hpp"
+
+namespace kagome::transaction_pool {
+
+  class TransactionPool {
+   public:
+    struct Status;
+    struct Limits;
+
+    virtual ~TransactionPool() = default;
+
+    /**
+     * Import one verified transaction to the pool. If it has unresolved
+     * dependencies (requires tags of transactions that are not in the pool
+     * yet), it will wait in the pool until its dependencies are solved, in
+     * which case it becomes ready and may be pruned, or it is banned from the
+     * pool for some amount of time as its longevity is reached or the pool is
+     * overflown
+     */
+    virtual outcome::result<void> submitOne(primitives::Transaction t) = 0;
+
+    /**
+     * Import several transactions to the pool
+     * @see submitOne()
+     */
+    virtual outcome::result<void> submit(
+        std::vector<primitives::Transaction> ts) = 0;
```

Comment:
maybe gsl::span?

Reply:
But in this case it wouldn't be a good idea to move them, which is as I remember done in the implementation?